### ISSUE TITLE
Dropdown: Deprecate custom content

### ISSUE THREAD

**Issue by interactivellama** [salesforce/design-system-react#1762]
Custom content for dropdown was a hack. If there's more than a listbox within a dropdown, then it should be a popover.

**Comment by zamhaq** [comment]
@interactivellama I would like to work on this. Can I get some more context? I understand the `custom-content` needs to be deprecated but I am sure what the statement - `it should be a popover` means?

**Comment by tanhengyeow** [comment]
@zamhaq I guess it means any custom content within a dropbox currently should be replaced with the [popover](https://www.lightningdesignsystem.com/components/popovers/) component. Still better for @interactivellama to clarify though :)

**Comment by zamhaq** [comment]
@interactivellama would be great if you can confirm so that I can proceed with the issue.

**Comment by interactivellama** [comment]
@zamhaq Yes, that is correct. Please use a `checkProps` warning with a deprecation of custom content. https://github.com/salesforce/design-system-react/blob/master/components/menu-dropdown/check-props.js#L1

See https://github.com/salesforce/design-system-react/blob/master/components/menu-dropdown/menu-dropdown.jsx#L165 for more info on children.

I think the way this would work is if any children are not a trigger or a list, then the warning would show in the console

```
	 * <Dropdown>
	 *   <Trigger>
	 *   <Button iconCategory="utility" iconName="settings" />
	 *   </Trigger>
	 *   <div>Look ma! This is Custom Content.</div>
	 *   <List options={[myArray]}/>
	 * </Dropdown>
```

Yes, please proceed. Apologies for the delay.

**Comment by interactivellama** [comment]
@tanhengyeow After reviewing your PR, I think modifying the documentation is enough--however, we should put in the menu-dropdown code comments about the prior use of custom content since it's all over the dropdown code. 

This would be a good place to transfer all the comments you removed and explain why it's there in the source, so that future contributors know why the logic is the way it is.
https://github.com/salesforce/design-system-react/blob/master/components/menu-dropdown/menu-dropdown.jsx#L850

After that I will merge.
